lib/driver.js

```diff
@@ -182,15 +185,38 @@ class YouiEngineDriver extends BaseDriver {
     let iosArgs = {
       javascriptEnabled: true,
     };
-    let iosdriver = new IOSDriver(iosArgs);
+    let driverName = "IOSDriver";
+    if (opts.platformVersion) {
+      let majorVer = opts.platformVersion.toString().split(".")[0];
+      if (parseInt(majorVer, 10) >= 10) {
+        driverName = "XCUITestDriver";
+      }
+    }
+
+    let iosdriver;
+    if (driverName === "XCUITestDriver") {
+      logger.info("Selected driver: XCUITestDriver");
+      iosdriver = new XCUITestDriver(iosArgs);
+    } else {
+      logger.info("Selected driver: IOSDriver");
+      iosdriver = new IOSDriver(iosArgs);
+    }
 
     let caps = {
       platformName: "iOS",
       browserName: "",
-      app: this.caps.app,
-      udid: this.caps.udid,
+      app: opts.app,
+      udid: opts.udid,
```

Comment:
Did you try different values here (including not setting one)? When I gave this a quick look previously, the Appium XCUITest driver didn't like if I didn't specify the `udid` in my caps (using simulator so `udid` wasn't needed).